src/languageservice/parser/jsonParser07.ts

```diff
@@ -1218,3 +1257,28 @@ function validate(
     return bestMatch;
   }
 }
+
+function getSchemaSource(schema: JSONSchema, originalSchema: JSONSchema): string | undefined {
+  if (schema) {
+    let label: string;
+    if (schema.title) {
+      label = schema.title;
+    } else if (originalSchema.title) {
+      label = originalSchema.title;
+    } else {
+      const uriString = schema.url ?? originalSchema.url;
+      if (uriString) {
+        const url = URI.parse(uriString);
+        if (url.scheme === 'file') {
+          label = url.fsPath;
```

Comment:
In this case won't label always just end up being whatever url.toString() is?

Reply:
Yes, it is done for local files, I assume the path `/home/user/folder/schema.json` will be more informative than `file:///home/user/folder/schema.json`. If URI has different schema than it used as is.